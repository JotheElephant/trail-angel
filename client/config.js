export const secrets = {
  wunderground: {
    apiKey: '42dec93e32e6a8d2'
  },

  yelp: {
    consumerKey: 'Sv6p3J6fwLEoFgvX8utCqg',
    consumerSecret: 'V2brLe04yNh8RYTBSZUNYul0bg0',
    token: 'OLPR1KSKhBoZ-M1k0vitBLer_dBEtX44',
    tokenSecret: 'pcrkmZkSCDJhjzzkMVt25Ip61j8'
  }
};

export const paths = {
  trailAngel: {
    baseUrl: 'http://localhost:4000'
  }
};

export const auth0Credentials = {
  clientId: "y2T9TOKA9Cq2bXarw3ZM1gZ5DXNBk1UV",
  domain: "trailangel.auth0.com"
};