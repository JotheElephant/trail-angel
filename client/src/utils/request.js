const request = {
  show: (url, id) => {
    return fetch(`${url}/${id}`)
      .then((response) => response.json())
      .then((responseJSON) => {
        return responseJSON;
      })
      .catch((err) => {
        console.error(err);
      });
  },

  get: (url, options = {}) => {
    return fetch(url, options)
      .then((response) => response.json())
      .then((responseJSON) => {
        return responseJSON;
      })
      .catch((err) => {
        console.error(err);
      });
  },

  add: (url, options) => {
    return fetch(url, {
      method: 'POST',
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(options)
    })
      .catch((err) => {
        console.error(err);
      });
  },

  update: () => {
    return fetch(url, {
      method: 'PUT',
      body: options
    })
      .then((response) => response.json())
      .then((responseJSON) => {
        return responseJSON;
      })
      .catch((err) => {
        console.error(err);
      });
  },

  remove: () => {
    return fetch(url, {
      method: 'DELETE',
      body: options
    })
      .then((response) => response.json())
      .then((responseJSON) => {
        return responseJSON;
      })
      .catch((err) => {
        console.error(err);
      });
  }
};

export default request;
