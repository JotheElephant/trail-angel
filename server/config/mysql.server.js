var Sequelize = require('sequelize');
const config = require('./environment/index');

console.log(JSON.stringify(config));
//var db = new Sequelize(config.MYSQL_DATABASE_URL);

//Use this instead of above db connection for now while working on local backend
var db = new Sequelize('trailangel', 'root', null, {
  host: 'localhost',
  dialect: 'mysql'
});


db.authenticate()
  .then(function() {
    console.log('Connection has been established successfully.');
  }, function (err) {
    console.error('Unable to connect to the database:', err);
  });

exports.db = db;
