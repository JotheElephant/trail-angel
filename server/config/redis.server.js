var redis = require('redis');
var bluebird = require('bluebird');
bluebird.promisifyAll(redis.RedisClient.prototype);
bluebird.promisifyAll(redis.Multi.prototype);
const config = require('./environment/index');
var geoMapper = redis.createClient(config.REDIS_PORT, config.REDIS_HOST);

geoMapper.on('connect', function() {
  console.log('Redis server is connected');
});

exports.geoMapper = geoMapper;
