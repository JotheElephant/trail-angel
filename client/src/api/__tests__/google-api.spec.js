'use strict';

import google from '../google-api';
import request from '../../utils/request';
import { secrets } from '../../../config';
import geolocationData from '../../../__tests__/fixtures/geolocation-data';

jest.mock('../../utils/request', () => {
  const distanceData = require('../../../__tests__/fixtures/distance-data').default;
  const reverseGeoLookupData
    = require('../../../__tests__/fixtures/reverse-geo-lookup-data').default;
  const elevationData
    = require('../../../__tests__/fixtures/elevation-data').default;

  return {
    get: jest.fn((url) => {
      if (url.indexOf('distancematrix') !== -1) {
        return Promise.resolve(distanceData);
      } else if (url.indexOf('geocode') !== -1){
        return Promise.resolve(reverseGeoLookupData);
      } else if (url.indexOf('elevation') !== -1) {
        return Promise.resolve(elevationData);
      }

    })
  };
});

describe('google api module', () => {
  it('returns distance data', () => {
    const origin = {
      latitude: geolocationData.coords.latitude,
      longitude: geolocationData.coords.longitude
    };

    const destination = {
      latitude: geolocationData.coords.latitude + 1,
      longitude: geolocationData.coords.longitude + 1
    };

    const url = `https://maps.googleapis.com/maps/api/distancematrix/` +
                `json?units=imperial&origins=${origin.latitude},` +
                `${origin.longitude}&destinations=${destination.latitude},` +
                `${destination.longitude}&key=${secrets.google.apiKey}`;

    return google.getDistance2Points(origin, destination)
      .then((results) => {
        expect(request.get).toBeCalledWith(url);
        expect(results).toMatchSnapshot();
      });
  });

  it('returns the city', () => {
    const coords = {
      latitude: geolocationData.coords.latitude,
      longitude: geolocationData.coords.longitude
    };

    const url = `https://maps.googleapis.com/maps/api/geocode/json?` +
                `latlng=${coords.latitude},${coords.longitude}` +
                `&key=${secrets.google.apiKey}`

    return google.getCity(coords)
      .then((results) => {
        expect(request.get).toBeCalledWith(url);
        expect(results).toMatchSnapshot();
      });

  });

  it('returns the elevation', () => {
    const coordinateSeriesData
      = require('../../../__tests__/fixtures/coordinate-series-data').default;
    const url = 'https://maps.googleapis.com/maps/api/elevation/json?locations=';

    return google.getElevation(coordinateSeriesData)
      .then((results) => {
        expect(request.get).toBeCalledWith(expect.arrayContaining([url]));
        expect(results).toMatchSnapshot();
      });
  });
});