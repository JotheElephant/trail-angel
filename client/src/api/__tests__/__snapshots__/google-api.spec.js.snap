exports[`google api module returns distance data 1`] = `
Object {
  "text": "7.1 mi",
  "value": 11414,
}
`;

exports[`google api module returns the city 1`] = `"Rhododendron"`;

exports[`google api module returns the elevation 1`] = `188.443359375`;
